src/vm/interoputil.cpp

```diff
@@ -2335,14 +2335,11 @@ VOID EnsureComStarted(BOOL fCoInitCurrentThread)
 
     if (g_fComStarted == FALSE)
     {
-        FinalizerThread::GetFinalizerThread()->SetRequiresCoInitialize();
-
-        // Attempt to set the thread's apartment model (to MTA by default). May not
-        // succeed (if someone beat us to the punch). That doesn't matter (since
-        // COM+ objects are now apartment agile), we only care that a CoInitializeEx
-        // has been performed on this thread by us.
+        FinalizerThread::GetFinalizerThread()->SetRequiresCoInitialize(); 
+        // Setting the thread's apartment state to Unknown when no attribute is supplied by the User.
+        // Thread is not CoInitialized
         if (fCoInitCurrentThread)
-            GetThread()->SetApartment(Thread::AS_InMTA, FALSE);
+            GetThread()->SetApartment(Thread::AS_Unknown, FALSE);
```

Comment:
Calling `SetApartment(Thread::AS_Unknown` will cause the COM to be uninitialized: https://github.com/dotnet/coreclr/blob/ddab65b2643ec2c3b7a376f4f44a05073124b732/src/vm/threads.cpp#L5168

This means that this EnsureComStarted method is doing opposite to what its name suggests. It is ensuring that the COM is not started, not that the COM is started.

Reply:
The dummy apartment state on Unix is handled in C# . It does not even go to here.